python-package/lightgbm/basic.py

```diff
@@ -0,0 +1,1189 @@
+            out_num_total_model = ctypes.c_int64(0)
+            _safe_call(_LIB.LGBM_BoosterCreateFromModelfile(
+                c_str(model_file), 
+                ctypes.byref(out_num_total_model),
+                ctypes.byref(self.handle)))
+            self.__num_total_model = out_num_total_model.value
+            tmp_out_len = ctypes.c_int64(0)
+            _safe_call(_LIB.LGBM_BoosterGetNumClasses(
+                self.handle,
+                ctypes.byref(tmp_out_len)))
+            self.num_class = tmp_out_len.value
+        elif booster_handle is not None:
+            self.__is_manage_handle = is_manage_handle
+            self.handle = booster_handle
+            tmp_out_len = ctypes.c_int64(0)
+            _safe_call(_LIB.LGBM_BoosterGetNumClasses(
+                self.handle,
+                ctypes.byref(tmp_out_len)))
+            self.num_class = tmp_out_len.value
+            _safe_call(_LIB.LGBM_BoosterGetCurrentIteration(
+                self.handle,
+                ctypes.byref(tmp_out_len)))
+            self.__num_total_model = self.num_class * tmp_out_len.value
+        else:
+            raise TypeError('Need Model file to create a booster')
+
+    def __del__(self):
+        if self.__is_manage_handle:
+            _safe_call(_LIB.LGBM_BoosterFree(self.handle))
+
+
+    def predict(self, data, num_iteration=-1, raw_score=False, pred_leaf=False, data_has_header=False, is_reshape=True):
+        """
+        Predict logic
+
+        Parameters
+        ----------
+        data : string/numpy array/scipy.sparse
+            Data source for prediction
+            When data is string type, it represents the path of txt file,
+        num_iteration : 
+            used iteration for prediction
+        raw_score : bool 
+            True for predict raw score
+        pred_leaf : bool
+            True for predict leaf index
+        data_has_header : bool
+            Used for txt data
+        is_reshape : bool
+            True for reshape to [nrow, ...] 
+
+        Returns
+        -------
+        Prediction result
+        """
+        if isinstance(data, Dataset):
+            raise TypeError("cannot use Dataset instance for prediction, please use raw data instead")
+        predict_type = C_API_PREDICT_NORMAL
+        if raw_score:
+            predict_type = C_API_PREDICT_RAW_SCORE
+        if pred_leaf:
+            predict_type = C_API_PREDICT_LEAF_INDEX
+        int_data_has_header = 0
+        if data_has_header:
+            int_data_has_header = 1
+        if is_str(data):
+            tmp_pred_fname = tempfile.NamedTemporaryFile(prefix="lightgbm_tmp_pred_").name
+            _safe_call(_LIB.LGBM_BoosterPredictForFile(
+                self.handle,
+                c_str(data), 
+                int_data_has_header,
+                predict_type,
+                num_iteration,
+                c_str(tmp_pred_fname)))
+            lines = open(tmp_pred_fname,"r").readlines()
```

Comment:
this file should be very small.
And i didn't give any reference to this file, so it will be destructed and closed after finished readlines.